### ISSUE TITLE
tf.keras.preprocessing.image.apply_affine_transform shifts tx as height and ty as width

### ISSUE THREAD

**Issue by maju116** [tensorflow/tensorflow#41460]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 20.04 LTS
- TensorFlow version (use command below): 2.2.0
- Python version: 3.x
- CUDA/cuDNN version: 10.1
- GPU model and memory: GeForce GTX 1070 8GB

**Describe the current behavior**
When shifting the image on x axis using `apply_affine_transform` output is shifted on y axis (and vice versa).

![00137](https://user-images.githubusercontent.com/11685462/87704058-11761b80-c79c-11ea-8b07-2ab05fe88fa7.jpg)

```
import tensorflow as tf
import cv2
import matplotlib.pyplot as plt
img = cv2.imread('00137.jpg')
img2 = tf.keras.preprocessing.image.apply_affine_transform(img, tx = 100, row_axis=0, col_axis=1)
plt.imshow(img2)
```

![Screenshot from 2020-07-16 19-14-43](https://user-images.githubusercontent.com/11685462/87704113-29e63600-c79c-11ea-8b33-8644a83ddc4e.png)

Switch in row_axis/col_axis is not doing anything.

**Describe the expected behavior**
Setting `tx` should shift image along `x` axis

**Standalone code to reproduce the issue**
Code in the example above

**Comment by Dynmi** [comment]
+1 
somehow confused....

**Comment by dgwozdz** [comment]
+1

**Comment by tanzhenyu** [comment]
Use `tf.keras.layers.experimental.preprocessing.RandomTranslation` instead

**Comment by maju116** [comment]
@tanzhenyu I will check it out, but I need non-random translation

**Comment by tanzhenyu** [comment]
> @tanzhenyu I will check it out, but I need non-random translation

you can definitely achieve non-random behavior by setting lower bound equals upper bound.

**Comment by saikumarchalla** [comment]
@maju116  Any update on this issue?  Please let us know if you need  any additional information.Thanks!

**Comment by jvishnuvardhan** [comment]
@maju116 As suggested by @tanzhenyu this was resolved by using keras preprocessing layers (`tf.keras.layers.experimental.preprocessing.RandomTranslation`). [Here](https://colab.research.google.com/gist/jvishnuvardhan/ab12d50bfc732df2c6e45824f28a9cb0/untitled.ipynb) is a gist for your reference. 

In the example, I have used width_factor as `0.2` which is same as (0.2, 0.2). As mentioned by @tanzhenyu when lower_bound and upper_bound are equal, there there is no random behavior introduced in the operation. 

![car_translated](https://user-images.githubusercontent.com/46058173/120564178-36c54580-c3bf-11eb-8625-9dfd0a1a693a.png)

Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by Tirbo06** [comment]
# SOLUTION
As mentioned in this thread [https://github.com/tensorflow/tensorflow/issues/39237#issuecomment-784553723](url)
The default arguments work on 4D images. If you want it to word in 3D images, use the argument channel_axis=2

**Comment by StanislawAntol** [comment]
I also ran into this issue on TF 2.5, which is using an outdated version of keras-preprocessing (i.e., v1.1.2). This issue was fixed via this [PR](https://github.com/keras-team/keras-preprocessing/pull/318), but had not made its way into the current TF versions. Though it seems like it will arrive soon based on [this commit](https://github.com/keras-team/keras/commit/373ad97c72ed1ac4b6898e85b2cfd7b016e4b469) from a month ago.
